app/sequencescape/sequencescape/api/v2/plate.rb

```diff
@@ -103,14 +105,16 @@ def size
     number_of_rows * number_of_columns
   end
 
-  def stock_plates(purpose_names: SearchHelper.stock_plate_names)
-    @stock_plates ||= stock_plate? ? [self] : ancestors.where(purpose_name: purpose_names)
+  def fetch_stock_plate_ancestors(purpose_names: SearchHelper.stock_plate_names)
+    ancestors.where(purpose_name: purpose_names)
   end
 
   def stock_plate
     return self if stock_plate?
 
-    stock_plates.order(id: :asc).last
+    # Note that it's only when we call last that we get an actual object to cache.
+    # If we cache the query to the API, then it will still be made every time we call stock_plate.
+    @stock_plate ||= fetch_stock_plate_ancestors.order(id: :asc).last
```

Comment:
@andrewsparkes and @StephenHulme This is the fix for the repeated requests for stock plates.  I've added a comment to make it clearer to future maintainers, but basically we were caching the query object back in the `stock_plates` method and not the stock plate itself.  The query object is cheap to create and hasn't made a request to the API at the time we were caching it.  Instead we need to cache the response which is only obtained after you call `last` on this line.  This gives us the speed up we need AFAIK.